consumer/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/consumer/service/saker/SakerServiceImpl.kt

```diff
@@ -0,0 +1,193 @@
+    @Autowired
+    private lateinit var standardKodeverk: StandardKodeverk
+    @Autowired
+    private lateinit var behandleHenvendelsePortType: BehandleHenvendelsePortType
+    @Autowired
+    private lateinit var arbeidOgAktivitet: ArbeidOgAktivitet
+    @Autowired
+    private lateinit var psakService: PsakService
+
+    private lateinit var arenaSaker: ArenaSaker
+    private lateinit var bidragSaker: BidragSaker
+    private lateinit var generelleSaker: GenerelleSaker
+    private lateinit var gsakSaker: GsakSaker
+    private lateinit var oppfolgingsSaker: OppfolgingsSaker
+    private lateinit var pensjonSaker: PensjonSaker
+
+    @PostConstruct
+    fun setup() {
+        arenaSaker = ArenaSaker(arbeidOgAktivitet)
+        bidragSaker = BidragSaker()
+        generelleSaker = GenerelleSaker()
+        gsakSaker = GsakSaker(sakV1, behandleSakWS)
+        oppfolgingsSaker = OppfolgingsSaker()
+        pensjonSaker = PensjonSaker(psakService)
+
+    }
+
+    override fun hentSaker(fnr: String): SakerService.Resultat {
+        val (gsakSaker, pesysSaker) = inParallel(
+                { hentSammensatteSakerResultat(fnr) },
+                { hentPensjonSakerResultat(fnr) }
+        )
+
+        val resultat = slaSammenGsakPesysSaker(gsakSaker, pesysSaker)
+        SakerUtils.leggTilFagsystemnavnOgTemanavn(resultat.saker, gsakKodeverk.hentFagsystemMapping(), standardKodeverk)
+
+        return resultat
+    }
+
+    override fun hentSammensatteSaker(fnr: String): List<Sak> {
+        return hentSammensatteSakerResultat(fnr).saker
+    }
+
+    override fun hentPensjonSaker(fnr: String): List<Sak> {
+        return hentPensjonSakerResultat(fnr).saker
+    }
+
+    fun hentSammensatteSakerResultat(fnr: String): SakerService.Resultat {
+        val resultat = SakerService.Resultat()
+        resultat.leggTilDataFraKilde(fnr, gsakSaker)
+        resultat.leggTilDataFraKilde(fnr, arenaSaker)
+        resultat.leggTilDataFraKilde(fnr, generelleSaker)
+        resultat.leggTilDataFraKilde(fnr, oppfolgingsSaker)
+        resultat.leggTilDataFraKilde(fnr, bidragSaker)
+
+        SakerUtils.leggTilFagsystemnavnOgTemanavn(resultat.saker, gsakKodeverk.hentFagsystemMapping(), standardKodeverk)
+
+        return resultat
+                .fjernIkkeGodkjenteSaker()
+    }
+
+    private fun SakerService.Resultat.fjernIkkeGodkjenteSaker(): SakerService.Resultat {
+        return SakerService.Resultat(
+                this.saker.filter(GODKJENT_FAGSAK or GODKJENT_GENERELL),
+                this.feiledeSystemer
+        )
+    }
+
+    fun hentPensjonSakerResultat(fnr: String): SakerService.Resultat {
+        val resultat = SakerService.Resultat().leggTilDataFraKilde(fnr, pensjonSaker);
+        SakerUtils.leggTilFagsystemnavnOgTemanavn(resultat.saker, gsakKodeverk.hentFagsystemMapping(), standardKodeverk)
+        return resultat
+    }
+
+    override fun knyttBehandlingskjedeTilSak(fnr: String?, behandlingskjede: String?, sak: Sak, enhet: String?) {
```

Comment:
fnr og behandlingskjede kan ikke være optional her tror jeg

Reply:
Sånn egentlig helt enig, men er litt java-kotlin-interop greier som stopper oss fra å gjøre det her tror jeg. :( 

F.eks om `enhet` er null så vil man få ett `EnhetIkkeSatt`-exception som fanges opp av `JournalforingController.java` for å gi en god feilmelding til bruker. Om vi setter alle feltene til å være non-optional, så vil man få generelle exceptions alà `java.lang.IllegalArgumentException: Parameter specified as non-null is null`. Og dette kan kan ikke skille fra at f.eks `fnr` var `null`.

Om vi kommer oss til det punktet hvor `SakerService` og alle som bruker klassen også er implementert i kotlin så vil vi mest sannsynlig kunne gjøre disse feltene til non-optional. Men per nå tror jeg egentlig alle parameterene burde være optional. :(